amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnector.scala

```diff
@@ -9,12 +9,29 @@ import com.rabbitmq.client._
 
 import scala.util.control.NonFatal
 
+private[amqp] case class ConnectionWithShutdown(conn: Connection, onShutdown: ShutdownListener) {
```

Comment:
I'm not convinced that the shutdown listeners belong here since there are stage-specific.

Reply:
Perhaps this syntax is misleading. This is effectively just a tuple to keep the fetched connection and its associated shutdown listener together so that they can be closed together. The listener can't be unregistered without the connection is was registered on so this construct felt necessary.